lib/core/extensions/stream.dart

```diff
@@ -0,0 +1,27 @@
+import 'dart:async';
+
+extension StreamExtensions<T> on Stream<T> {
+  Stream<T> handleOnlyException() {
+    final controller = StreamController<T>();
+    listen((event) {
+      controller.sink.add(event);
+    }, onError: (error) {
+      if (error is! Error) {
+        controller.sink.addError(error as Object);
```

Comment:
I'm seeing that from some of your sequent implementation, the downstreams are usually taking an Object from the error, which isn't a nice idea imo. I think it's always better to know the possible class type to handle rather than broadly sending a master class Object. At least an Exception or an extended class from Error/Exception then wrap this in, then delegate it to the downstream would be cleaner to handle imo.

Reply:
@markgravity ah, if you can resolve this issue, it's wonderful